go-controller/pkg/node/secondary_node_network_controller.go

```diff
@@ -38,11 +43,29 @@ func NewSecondaryNodeNetworkController(cnnci *CommonNodeNetworkControllerInfo, n
 func (nc *SecondaryNodeNetworkController) Start(ctx context.Context) error {
 	klog.Infof("Start secondary node network controller of network %s", nc.GetNetworkName())
 
-	handler, err := nc.watchPodsDPU()
-	if err != nil {
-		return err
+	// enable adding ovs ports for dpu pods in both primary and secondary user defined networks
+	if (config.OVNKubernetesFeature.EnableMultiNetwork || util.IsNetworkSegmentationSupportEnabled()) && config.OvnKubeNode.Mode == types.NodeModeDPU {
+		handler, err := nc.watchPodsDPU()
```

Comment:
so we will use the VF rep for the UDN primary network, but at the same time the default controller will try to use the VF rep for the default cluster network. In the default path, we need to not plug the VF rep and use a veth if possible, since its just slow path for kubernetes healthchecks. @girishmg I'm not sure if it is possible to plumb a veth across host -> DPU for default cluster network access. Can you help us out here? I wanted to avoid consuming 2 VF reps for pods with a UDN.